uwblueprint/planet-read
# Implement get all story translations query for admins

## Notion ticket link



[Implement get all story translation query for admins](https://www.notion.so/uwblueprintexecs/Implement-get-all-story-translation-query-for-admins-1d76ccff6e9d4628a1f46a145e748ac7)



## Implementation description

- Modified existing `get_story_translations` function in `story_service.py` -> `get_story_translations_by_user`
- Added in optional params to filter story translations by (level, language, stage, title)



## Steps to test

1. Double check existing `storyTranslationsByUser` [query](http://localhost:5000/graphql?query=%7B%0A%20%20storyTranslationsByUser(userId%3A4%2C%20isTranslator%3A%20true)%20%7B%0A%20%20%20%20title%0A%20%20%20%20description%0A%20%20%20%20level%0A%20%20%20%20language%0A%20%20%20%20stage%0A%20%20%20%20translatorId%0A%20%20%20%20reviewerId%0A%20%20%7D%0A%7D%0A) still works as expected
2. Check that you are unable to make a [query](http://localhost:5000/graphql?query=%7B%0A%20%20storyTranslations%20(language%3A%22GERMAN%22%2C%20level%3A%203%2C%20storyTitle%3A%22Mockingbird%22%2C%20stage%3A%20%0A%22TRANSLATE%22)%20%7B%0A%20%20%20%20title%0A%20%20%20%20description%0A%20%20%20%20level%0A%20%20%20%20language%0A%20%20%20%20stage%0A%20%20%20%20translatorId%0A%20%20%20%20reviewerId%0A%20%20%7D%0A%7D%0A) to `storyTranslations` without logging in first
3. Login as an admin and retry the above [query](http://localhost:5000/graphql?query=%7B%0A%20%20storyTranslations%20(language%3A%22GERMAN%22%2C%20level%3A%203%2C%20storyTitle%3A%22Mockingbird%22%2C%20stage%3A%20%0A%22TRANSLATE%22)%20%7B%0A%20%20%20%20title%0A%20%20%20%20description%0A%20%20%20%20level%0A%20%20%20%20language%0A%20%20%20%20stage%0A%20%20%20%20translatorId%0A%20%20%20%20reviewerId%0A%20%20%7D%0A%7D%0A)
```
{
  "data": {
    "storyTranslations": [
      {
        "title": "To Kill a Mockingbird",
        "description": "To Kill a Mockingbird is a novel by the American author Harper Lee. A sentence cannot do this novel justice.",
        "level": 3,
        "language": "GERMAN",
        "stage": "TRANSLATE",
        "translatorId": 4,
        "reviewerId": null
      }
    ]
  }
}
```
4. Test different queries while changing the parameter values/removing certain parameters
5. Try passing in [no parameters](http://localhost:5000/graphql?query=%7B%0A%20%20storyTranslations%20%7B%0A%20%20%20%20id%0A%20%20%7D%0A%7D) and verify that you see 13 story translations



## What should reviewers focus on?

- `story_service.py`
- did we want the `story_title` filter to be an exact match or is like okay?

## Checklist
